src/YoutubeDl.php

```diff
@@ -116,6 +122,77 @@ public function download(string $url): Video
         return $this->processDownload($process);
     }
 
+    public function downloadAsync(string $url, LoopInterface $loop): PromiseInterface
+    {
+        if (!class_exists('LoopInterface')) {
+            throw new \RuntimeException('React event loop package not installed.');
+        }
+
+        if (!class_exists('Deferred')) {
+            throw new \RuntimeException('React promise package not installed.');
+        }
+
+        $deferred = new Deferred();
+        $promise = $deferred->promise();
+
+        try {
+            if (!$this->downloadPath) {
+                throw new \RuntimeException('No download path was set.');
+            }
+
+            if (!$this->isUrlSupported($url)) {
+                throw new UrlNotSupportedException(sprintf('Provided url "%s" is not supported.', $url));
+            }
+
+            $arguments = [
```

Comment:
Maybe would be nice to make a private helper to share download process building?